docs_src/design/adr/device-service/0021-Profile-Changes.md

```diff
@@ -0,0 +1,116 @@
+| Manufacturer | string | Anytime | As this property does not affect any functionality other than profile query, this property can be changed at any time – to include setting the string to an empty string. |
+| Description | string | Anytime | As this property does not affect any functionality other than profile query, this property can be changed at any time – to include setting the string to an empty string. |
+| Model | string | Anytime | As this property does not affect any functionality other than profile query, this property can be changed at any time – to include setting the string to an empty string. |
+| Labels | string array | Anytime | As this property does not affect any functionality other than profile query, this property can be changed at any time – to include adding, removing or modifying any of the strings in the array. |
+| Device Resources | array | Situational | When the device resource is not associated to any reading or event, the device resource list can be modified (i.e., adding or removing the associated device resources).  Once the device resource is associated to a reading or event (directly or indirectly via device command), only new resources can be added to the array.  Deleting is prohibited.|
+| Device Commands | array | Situational | When the device command is not associated to any reading or event, the device commands list can be modified (i.e., adding or removing the associated device commands).  Once the device command is associated to a reading or event, only new commands can be added to the array.  Deleting is prohibited.  Device commands are optional and so the array can always be empty. |
+| *Device Resource Properties* |||
+| Description | string | Anytime | As this property does not affect any functionality, this property can be changed at any time – to include setting the string to an empty string. |
+| Name | string | Situational | When the device profile is not associated to any reading or event, this field can be modified.  It can never be blank and must be unique across the entire profile. |
+| IsHidden | bool | Anytime | This field allows the device resource to be exposed (default) via the command service.  This change does not affect the behavior but only visibility.  This field is optional but is false by default when not provided. |
+| Tag | string | Situational | As this property does not affect any functionality, this property can be changed at any time – to include setting the string to an empty string. |
+| Properties | Resource Properties | Situational | When the device resource (directly or indirectly through device command) is not associated to any reading or event, this field can be modified.  Because the ValueType and ReadWrite properties of this field are required, it can never be empty. |
```

Comment:
I think you can leave out the "(directly or indirectly through device command)". Readings always specify device resources. Events can specify both.